### PR TITLE
Migrate form to data sources

### PR DESCRIPTION
**Pull request by TrySound**
Ref https://github.com/webstudio-is/webstudio-builder/issues/1529

Instead of SuccessMessage and ErrorMessage components we can use Box with bound state from form.

For now this is only boolean state which should be set on Form component side. Soon we explore using expression as data sources to change only enum value instead of booleans.

Changes in template
- ErrorMessage and SuccessMessage are replaced with Box with "show" bound to expression
- ErrorMessage and SuccessMessage have text in own children
- the rest of Form content is wrapped with Box bound to expression
- Form got state prop bound to variable
- Form initialState prop is deleted

Migration just bind show props of all from direct children to data sources. Without additional wrapping.

### CODE REVIEW

File: packages/sdk-components-react-remix/src/form.ws.tsx [hunk 1]
```diff
@@ -30,59 +32,103 @@ export const meta: WsComponentMeta = {
...
+              component: "Button",
+              children: [{ type: "text", value: "Submit" }],
             },
           ],
         },
+
         {
           type: "instance",
-          component: "ErrorMessage",
+          label: "SuccessMessage",
```
Comment (kof) [inline packages/sdk-components-react-remix/src/form.ws.tsx:96]: any reason to use camel case instead  of 2 capitalized words?

File: packages/sdk-components-react-remix/src/form.ws.tsx [hunk 2]
```diff
@@ -30,59 +32,103 @@ export const meta: WsComponentMeta = {
...
+              value: false,
+              dataSourceRef: {
+                type: "expression",
+                name: "formSuccess",
+                code: `formState === 'success'`,
+              },
+            },
+          ],
           children: [
+            { type: "text", value: "Thank you for getting in touch!" },
```
Comment (kof) [inline packages/sdk-components-react-remix/src/form.ws.tsx:111]: seeems like this value is not rendered

File: packages/sdk-components-react-remix/src/form.ws.tsx [hunk 3]
```diff
@@ -30,59 +32,103 @@ export const meta: WsComponentMeta = {
...
+              type: "boolean",
+              value: false,
+              dataSourceRef: {
+                type: "expression",
+                name: "formError",
+                code: `formState === 'error'`,
+              },
             },
           ],
+          children: [{ type: "text", value: "Sorry, something went wrong." }],
```
Comment (kof) [inline packages/sdk-components-react-remix/src/form.ws.tsx:131]: same with this one, doesn't render the value

File: packages/sdk-components-react-remix/src/form.ws.tsx (hunk 1 above)
Comment (TrySound) [reply packages/sdk-components-react-remix/src/form.ws.tsx:96]: I guess no just mimicking previous names

**Review by kof — APPROVED** [review]

**Comment by kof** [comment]
Tested locally, seems to work, it didn't wrap contents of the form into a box with this migration but I think this is what you expected
Screenshot 2023-07-02 at 12 53 02

**Review by istarkov — APPROVED** [review]

**Pull request merged**